StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs

```diff
@@ -107,9 +156,141 @@ public void Test()
+        Action<int> c = x => { Console.WriteLine(); };
+        Action<int, int> d = (x, y) => { Console.WriteLine(); };
+        Action<int, int> e = delegate (int x, int y = 0) { Console.WriteLine(); };
+        Action<int, int> f = delegate (int x, [Obsolete]int y) { Console.WriteLine(); };
+        Action<int, int> g = delegate (int x, params int y) { Console.WriteLine(); };
+        Action<int> h = delegate (int x, __arglist) { Console.WriteLine(); };
+    }
+}";
+            var expected = new[]
+            {
+                this.CSharpDiagnostic().WithLocation(8, 20),
+                this.CSharpDiagnostic().WithLocation(9, 20),
+                this.CSharpDiagnostic().WithLocation(10, 25),
+                this.CSharpDiagnostic().WithLocation(11, 30),
+                this.CSharpDiagnostic().WithLocation(12, 30),
+                this.CSharpDiagnostic(CS1065).WithLocation(12, 53),
+                this.CSharpDiagnostic().WithLocation(13, 30),
+                this.CSharpDiagnostic(CS7014).WithLocation(13, 47),
+                this.CSharpDiagnostic().WithLocation(14, 30),
+                this.CSharpDiagnostic(CS1670).WithLocation(14, 47),
+                this.CSharpDiagnostic().WithLocation(15, 25),
+                this.CSharpDiagnostic(CS1669).WithLocation(15, 42)
+            };
+
+            var expectedAfterFix = new[]
+            {
+                this.CSharpDiagnostic().WithLocation(12, 30),
+                this.CSharpDiagnostic(CS1065).WithLocation(12, 53),
+                this.CSharpDiagnostic().WithLocation(13, 30),
+                this.CSharpDiagnostic(CS7014).WithLocation(13, 47),
+                this.CSharpDiagnostic().WithLocation(14, 30),
+                this.CSharpDiagnostic(CS1670).WithLocation(14, 47),
+                this.CSharpDiagnostic().WithLocation(15, 25),
+                this.CSharpDiagnostic(CS1669).WithLocation(15, 42)
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpDiagnosticAsync(fixedCode, expectedAfterFix, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedCode, allowNewCompilerDiagnostics: true).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestSimpleDelegateUseWithTriviaAsync()
+        {
+            var testCode = @"
+using System;
+public class TypeName
+{
+    public void Test()
+    {
+        Action action1 = /*a*/delegate/*b*/{ };
+        Action action2 = /*a*/delegate/*b*/(/*c*/)/*d*/ { };
+        Action<int> action3 = /*a*/delegate/*b*/(/*c*/int/*d*/i/*e*/)/*f*/{ };
+    }
+}";
+
+            string fixedCode = @"
+using System;
+public class TypeName
+{
+    public void Test()
+    {
+        Action action1 = /*a*/()/*b*/ => { };
+        Action action2 = /*a*//*b*/(/*c*/)/*d*/ => { };
+        Action<int> action3 = /*a*//*b*//*c*/i/*d*//*e*//*f*/ => { };
```

Comment:
:thought_balloon: It might be nice to add a test case for `delegate(List</* c1 */ int> i)` for completeness.